tests/SprykerTest/Glue/AppPaymentBackendApi/Plugin/PaymentConfirmDisconnectionRequestValidatorPluginTest.php

```diff
@@ -0,0 +1,192 @@
+        $this->assertCount(1, $glueRequestValidationTransfer->getErrors());
+        $this->assertSame(
+            AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_TENANT_IDENTIFIER_MISSING,
+            $glueRequestValidationTransfer->getErrors()[0]->getCode(),
+        );
+    }
+
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsSuccessIfThereAreNoPaymentsForTenantIdentifierGivenInTheRequest(): void
+    {
+        // Arrange
+        $this->tester->setDependency(AppPaymentBackendApiDependencyProvider::FACADE_TRANSLATOR, $this->getTranslatorFacadeMock());
+
+        $paymentConfirmDisconnectionRequestValidatorPlugin = new PaymentConfirmDisconnectionRequestValidatorPlugin();
+        $glueRequestTransfer = new GlueRequestTransfer();
+        $glueRequestTransfer->setMeta([
+            GlueRequestPaymentMapper::HEADER_TENANT_IDENTIFIER => [static::TENANT_IDENTIFIER],
+        ]);
+
+        // Act
+        $glueRequestValidationTransfer = $paymentConfirmDisconnectionRequestValidatorPlugin
+            ->validate($glueRequestTransfer);
+
+        // Assert
+        $this->assertTrue($glueRequestValidationTransfer->getIsValid());
+    }
+
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsErrorIfThereArePaymentsForTenantIdentifierGivenInTheRequest(): void
+    {
+        // Arrange
+        $this->tester->setDependency(AppPaymentBackendApiDependencyProvider::FACADE_TRANSLATOR, $this->getTranslatorFacadeMock());
+
+        $paymentConfirmDisconnectionRequestValidatorPlugin = new PaymentConfirmDisconnectionRequestValidatorPlugin();
+        $glueRequestTransfer = new GlueRequestTransfer();
+        $glueRequestTransfer->setMeta([
+            GlueRequestPaymentMapper::HEADER_TENANT_IDENTIFIER => [static::TENANT_IDENTIFIER],
+        ]);
+
+        $this->tester->havePayment([
+            PaymentTransfer::TENANT_IDENTIFIER => static::TENANT_IDENTIFIER,
+        ]);
+
+        // Act
+        $glueRequestValidationTransfer = $paymentConfirmDisconnectionRequestValidatorPlugin
+            ->validate($glueRequestTransfer);
+
+        // Assert
+        $this->assertFalse($glueRequestValidationTransfer->getIsValid());
+        $this->assertCount(1, $glueRequestValidationTransfer->getErrors());
+        $this->assertSame(
+            AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_CANNOT_BE_PROCEEDED,
+            $glueRequestValidationTransfer->getErrors()[0]->getCode(),
+        );
+        $this->assertInstanceOf(GlueErrorConfirmTransfer::class, $glueRequestValidationTransfer->getErrors()[0]->getConfirm());
+    }
+
+    /**
+     * @dataProvider confirmationStatusDataProvider
+     *
+     * @param string $confirmationStatus
+     *
+     * @return void
+     */
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsErrorIfThereArePaymentsForTenantIdentifierGivenInTheRequestAndTheRequestContainsConfirmationCanceledResponse(
```

Comment:
Same here. Payments are not in the request but rather persisted.

Reply:
Rephrased.